src/types/local_date.rs

```diff
@@ -0,0 +1,268 @@
+
+    /// Returns the number of days since 01.01.1970 based on a String representing the date.
+    #[napi]
+    pub fn from_string(value: String) -> napi::Result<i32> {
+        match value.chars().filter(|c| *c == '-').count() {
+            d if d < 2 => match value.parse::<i32>() {
+                Ok(val) => Ok(val),
+                Err(_) => Err(js_error("Invalid format of string")),
+            },
+            2 | 3 => {
+                let lambda = |s: String| -> Result<(i8, i8, i32), ParseIntError> {
+                    let mut cut: usize = 0;
+                    if s.starts_with('-') {
+                        cut = 1;
+                    }
+
+                    let parts: Vec<&str> = s.get(cut..).unwrap().split('-').collect();
+                    Ok((
+                        parts[2].parse::<i8>()?,
+                        parts[1].parse::<i8>()?,
+                        match cut {
+                            0 => parts[0].parse::<i32>()?,
+                            _ => -parts[0].parse::<i32>()?,
+                        },
+                    ))
+                };
+
+                match lambda(value) {
+                    Ok(s) => Ok(Self::days_since_epoch(s.0, s.1, s.2)),
+                    Err(_) => Err(js_error("Invalid parse of string")),
+                }
+            }
+            _ => Err(js_error("Invalid format of string")),
+        }
+    }
+
+    fn check_valid_date(day: i8, month: i8, year: i32) -> napi::Result<Self> {
+        if !(1..=12).contains(&month) {
+            return Err(js_error("Invalid month"));
+        }
+
+        let days_in_month = Self::days_in_month(month, year);
+
+        if !(day >= 1 && day <= days_in_month) {
+            return Err(js_error("Invalid number of day"));
+        }
+
+        let value = Self::days_since_epoch(day, month, year);
+
+        Ok(LocalDateWrapper {
+            year,
+            month,
+            day,
+            value,
+            in_date: (MIN_JS_DATE..=MAX_JS_DATE).contains(&value),
+        })
+    }
+
+    pub fn get_cql_date(&self) -> CqlDate {
+        CqlDate(((1 << 31) + self.value) as u32)
+    }
+
+    pub fn from_cql_date(date: CqlDate) -> Self {
+        let value: i32 = date.0 as i32 - (1 << 31);
+        let mut my = LocalDateWrapper {
+            value,
+            year: 1970,
+            month: 1,
+            day: 1,
+            in_date: (MIN_JS_DATE..=MAX_JS_DATE).contains(&value),
+        };
+
+        if my.in_date {
+            my.add_day(value.into());
+        }
+        my
+    }
+}
+
+impl fmt::Display for LocalDateWrapper {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.in_date {
+            write!(
+                f,
+                "{}-{}-{}",
+                if self.year >= 1000 || self.year <= -1000 {
+                    format!("{}", self.year)
+                } else if self.year < 0 {
+                    let year_string = (-(self.year)).to_string();
```

Comment:
```suggestion
                    let year_string = (self.year.unsigned_abs()).to_string();
```
With the current logic we shouldn't reach the values where this is a problem, but is better to use safer way of doing it 😛. ~~We don't want to have someone open PR to this repo complaining about invalid handling of negative values~~